flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -0,0 +1,333 @@
+	}
+
+	private static MemorySize deriveManagedMemoryAbsoluteOrWithFraction(final Configuration config, final MemorySize base) {
+		if (isManagedMemorySizeExplicitlyConfigured(config)) {
+			return getManagedMemorySize(config);
+		} else {
+			return deriveWithFraction(base, getManagedMemoryRangeFraction(config));
+		}
+	}
+
+	private static MemorySize deriveShuffleMemoryWithFraction(final Configuration config, final MemorySize base) {
+		return deriveWithFraction(base, getShuffleMemoryRangeFraction(config));
+	}
+
+	private static MemorySize deriveShuffleMemoryWithInverseFraction(final Configuration config, final MemorySize base) {
+		return deriveWithInverseFraction(base, getShuffleMemoryRangeFraction(config));
+	}
+
+	private static MemorySize deriveJvmOverheadWithFraction(final Configuration config, final MemorySize base) {
+		return deriveWithFraction(base, getJvmOverheadRangeFraction(config));
+	}
+
+	private static MemorySize deriveJvmOverheadWithInverseFraction(final Configuration config, final MemorySize base) {
+		return deriveWithInverseFraction(base, getJvmOverheadRangeFraction(config));
+	}
+
+	private static MemorySize deriveWithFraction(final MemorySize base, final RangeFraction rangeFraction) {
+		final long relative = (long) (rangeFraction.fraction * base.getBytes());
+		return new MemorySize(Math.max(rangeFraction.minSize.getBytes(), Math.min(rangeFraction.maxSize.getBytes(), relative)));
+	}
+
+	private static MemorySize deriveWithInverseFraction(final MemorySize base, final RangeFraction rangeFraction) {
+		checkArgument(rangeFraction.fraction < 1);
+		final long relative = (long) (rangeFraction.fraction / (1 - rangeFraction.fraction) * base.getBytes());
+		return new MemorySize(Math.max(rangeFraction.minSize.getBytes(), Math.min(rangeFraction.maxSize.getBytes(), relative)));
+	}
+
+	private static MemorySize getFrameworkHeapMemorySize(final Configuration config) {
+		return MemorySize.parse(config.getString(TaskManagerOptions.FRAMEWORK_HEAP_MEMORY));
+	}
+
+	private static MemorySize getTaskHeapMemorySize(final Configuration config) {
+		checkArgument(isTaskHeapMemorySizeExplicitlyConfigured(config));
+		return MemorySize.parse(config.getString(TaskManagerOptions.TASK_HEAP_MEMORY));
+	}
+
+	private static MemorySize getTaskOffHeapMemorySize(final Configuration config) {
+		return MemorySize.parse(config.getString(TaskManagerOptions.TASK_OFF_HEAP_MEMORY));
+	}
+
+	private static MemorySize getManagedMemorySize(final Configuration config) {
+		checkArgument(isManagedMemorySizeExplicitlyConfigured(config));
+		return MemorySize.parse(config.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE));
+	}
+
+	private static RangeFraction getManagedMemoryRangeFraction(final Configuration config) {
+		final MemorySize minSize = new MemorySize(0);
+		final MemorySize maxSize = new MemorySize(Long.MAX_VALUE);
+		final double fraction = config.getFloat(TaskManagerOptions.MANAGED_MEMORY_FRACTION);
```

Comment:
What happens if the user has `LEGACY_MANAGED_MEMORY_FRACTION` configured? Don't we want to support backwards compatibility?

Reply:
True. I forgot to add `LEGACY_MANAGED_MEMORY_FRACTION.key()` as a deprecated key to the new `MANAGED_MEMORY_FRACTION`

Reply:
I see. Could we transform `legacyFraction` into `flip49Fraction` if `LEGACY_MANAGED_MEMORY_FRACTION` has been specified and `MANAGED_MEMORY_FRACTION` has not been specified?